pyramid/tierFS.go

```diff
@@ -84,93 +88,71 @@ func NewFS(c *Config) (FS, error) {
 // 1. Adds stored files to the eviction control
 // 2. Remove workspace directories and all its content if it
 //	  exist under the namespace dir.
-func handleExistingFiles(eviction *lruSizeEviction, fsLocalBaseDir string) error {
-	var workspaceDirs []string
+func handleExistingFiles(eviction eviction, fsLocalBaseDir string) error {
 	if err := filepath.Walk(fsLocalBaseDir, func(rPath string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
 		if info.IsDir() {
 			if info.Name() == workspaceDir {
 				// skipping workspaces and saving them for later delete
-				workspaceDirs = append(workspaceDirs, rPath)
+				if err := os.RemoveAll(rPath); err != nil {
+					return fmt.Errorf("removing dir: %w", err)
+				}
 				return filepath.SkipDir
 			}
 			return nil
 		}
 
-		if !eviction.store(relativePath(rPath), info.Size()) {
-			// Cache rejected the file, can be removed from local storage.
-			// Should be relatively rare, as cache sum cost on startup should accurately
-			// represent previous cost limitations.
-			removeFromLocal(fsLocalBaseDir, relativePath(rPath))
-		}
+		eviction.store(relativePath(rPath), info.Size())
 		return nil
 	}); err != nil {
 		return fmt.Errorf("walking the fs dir: %w", err)
 	}
 
-	for _, dir := range workspaceDirs {
-		if err := os.RemoveAll(dir); err != nil {
-			return fmt.Errorf("removing dir: %w", err)
-		}
-	}
-
 	return nil
 }
 
 func (tfs *TierFS) removeFromLocal(rPath relativePath) {
-	removeFromLocal(tfs.fsLocalBaseDir, rPath)
+	// This will be called by the cache eviction mechanism during entry insert.
+	// We don't want to wait while the file is being removed from the local disk.
+	go removeFromLocal(tfs.logger, tfs.fsLocalBaseDir, rPath)
 }
 
-func removeFromLocal(fsLocalBaseDir string, rPath relativePath) {
-	_ = os.Remove(path.Join(fsLocalBaseDir, string(rPath)))
-}
-
-// Store adds the local file to the FS.
-// on successful operation, file will no longer be available under the originalPath.
-func (tfs *TierFS) Store(namespace, originalPath, filename string) error {
-	if err := tfs.validateArgs(namespace, filename); err != nil {
-		return fmt.Errorf("invalid args: %w", err)
+func removeFromLocal(logger logging.Logger, fsLocalBaseDir string, rPath relativePath) {
+	p := path.Join(fsLocalBaseDir, string(rPath))
+	if err := os.Remove(p); err != nil {
+		logger.WithError(err).Errorf("Removing file %s", p)
```

Comment:
```suggestion
		logger.WithError(err).WithField("path", p).Errorf("failed to remove file")
```
Logs ideally have no formatting, everything goes in fields.  That allows me e.g. to estimate a histogram of eviction frequencies by file.